<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConfigDialog</class>
 <widget class="QDialog" name="ConfigDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>658</width>
    <height>483</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DbMaster - Preferences</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QListWidget" name="listWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>150</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="iconSize">
       <size>
        <width>24</width>
        <height>24</height>
       </size>
      </property>
      <property name="currentRow">
       <number>0</number>
      </property>
      <item>
       <property name="text">
        <string>General</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/img/configure.png</normaloff>:/img/configure.png</iconset>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Query editor</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/img/edit.png</normaloff>:/img/edit.png</iconset>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Plugins</string>
       </property>
      </item>
     </widget>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="page">
       <layout class="QFormLayout" name="formLayout">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Default driver</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="DbComboBox" name="defaultDbCombo">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_3">
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="1">
         <widget class="QTabWidget" name="tabWidget">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab_4">
           <attribute name="title">
            <string>General</string>
           </attribute>
           <layout class="QFormLayout" name="formLayout_2">
            <item row="0" column="0" colspan="2">
             <widget class="QCheckBox" name="edAutoSaveCheckBox">
              <property name="text">
               <string>Save query before execute it</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0" colspan="2">
             <widget class="QCheckBox" name="editorSemanticCheckBox">
              <property name="text">
               <string>Enable semantic analysis</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Default encoding</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QComboBox" name="encodingComboBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="currentIndex">
               <number>1</number>
              </property>
              <item>
               <property name="text">
                <string>latin1</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>utf8</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab">
           <attribute name="title">
            <string>Font &amp;&amp; Size</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0">
             <widget class="QGroupBox" name="groupBox">
              <property name="title">
               <string>Editor's font</string>
              </property>
              <layout class="QFormLayout" name="formLayout_3">
               <item row="0" column="0">
                <widget class="QLabel" name="label_2">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>&amp;Font</string>
                 </property>
                 <property name="buddy">
                  <cstring>editorFontCombo</cstring>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QFontComboBox" name="editorFontCombo">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_3">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>S&amp;ize</string>
                 </property>
                 <property name="buddy">
                  <cstring>editorFontSizeSpin</cstring>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSpinBox" name="editorFontSizeSpin">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>T&amp;ab size</string>
                 </property>
                 <property name="buddy">
                  <cstring>tabsizeSpin</cstring>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QSpinBox" name="tabsizeSpin">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>10</number>
                 </property>
                 <property name="value">
                  <number>4</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>You can type some text here :</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QueryTextEdit" name="editorTestArea">
              <property name="html" stdset="0">
               <string notr="true">&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:9pt;&quot;&gt;SELECT *&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:9pt;&quot;&gt;FROM SomeTable, now()&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:9pt;&quot;&gt;WHERE SomeField = 1&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:9pt;&quot;&gt;ORDER BY AnotherField ASC;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_2">
           <attribute name="title">
            <string>Syntax highlighting</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="1" column="0" colspan="2">
             <widget class="QGroupBox" name="shGroupBox">
              <property name="title">
               <string>Enable Syntax highlighting</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
              <layout class="QGridLayout" name="gridLayout_6">
               <item row="0" column="0">
                <widget class="QListWidget" name="shListWidget">
                 <property name="currentRow">
                  <number>-1</number>
                 </property>
                 <item>
                  <property name="text">
                   <string>Basic SQL keywords</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>SQL functions</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>SQL types</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Strings</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Numbers</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Comments</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Table name</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Field name</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QGroupBox" name="groupBox_2">
                 <property name="title">
                  <string/>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_4">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_5">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                    <property name="text">
                     <string>Color</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0" colspan="2">
                   <widget class="QCheckBox" name="shBoldButton">
                    <property name="font">
                     <font>
                      <weight>75</weight>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>Bold</string>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="0" colspan="2">
                   <widget class="QCheckBox" name="shItalicButton">
                    <property name="font">
                     <font>
                      <italic>true</italic>
                     </font>
                    </property>
                    <property name="text">
                     <string>Italic</string>
                    </property>
                   </widget>
                  </item>
                  <item row="3" column="0" colspan="2">
                   <spacer name="verticalSpacer_2">
                    <property name="orientation">
                     <enum>Qt::Vertical</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>20</width>
                      <height>40</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                  <item row="0" column="1">
                   <widget class="ColorButton" name="shColorButton">
                    <property name="text">
                     <string/>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_3">
           <attribute name="title">
            <string>Auto-completion</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="0" column="0">
             <widget class="QGroupBox" name="compGroubBox">
              <property name="title">
               <string>Enable auto-completion</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
              <layout class="QFormLayout" name="formLayout_4">
               <item row="0" column="0">
                <widget class="QLabel" name="label_7">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Show completion popup after</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <layout class="QHBoxLayout" name="horizontalLayout">
                 <item>
                  <widget class="QSpinBox" name="acSpinBox">
                   <property name="minimum">
                    <number>1</number>
                   </property>
                   <property name="value">
                    <number>3</number>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_8">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>characters</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>&lt;b&gt;Warning !&lt;/b&gt; All modifications concerning the editor will work after restarting DbMaster.</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_2"/>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Apply|QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DbComboBox</class>
   <extends>QComboBox</extends>
   <header>../src/widgets/dbcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QueryTextEdit</class>
   <extends>QPlainTextEdit</extends>
   <header>../src/widgets/querytextedit.h</header>
  </customwidget>
  <customwidget>
   <class>ColorButton</class>
   <extends>QToolButton</extends>
   <header>../src/widgets/colorbutton.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>listWidget</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>289</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
